drivers/adc/ad463x/ad463x.c

```diff
@@ -529,12 +665,100 @@ int32_t ad463x_init(struct ad463x_dev **device,
 	axi_clkgen_remove(dev->clkgen);
 error_gpio:
 	no_os_gpio_remove(dev->gpio_resetn);
+	no_os_gpio_remove(dev->gpio_pgia_a0);
+	no_os_gpio_remove(dev->gpio_pgia_a1);
 error_dev:
 	no_os_free(dev);
 
 	return -1;
 }
 
+/**
+ * @brief Find closest value of an array of intergers.
+ * @param target - Target/reference value.
+ * @param array - Input Array.
+ * @param size - array size.
+ * @return Gain control index, in case of success, negative number otherwise
+ */
+static int32_t ad463x_find_closest(int32_t target, const int32_t array[],
```

Comment:
this could be moved to no_os_util

Reply:
Actualy I found an existing implementation of that. i will check that